src/components/Thumbnail.js

```diff
@@ -0,0 +1,53 @@
+import classnames from 'classnames';
+import { toChildArray } from 'preact';
+
+import { Spinner } from './Spinner';
+
+/**
+ * @typedef {import('preact').ComponentChildren} Children
+ *
+ * @typedef ThumbnailProps
+ * @prop {Children|null} [children] - Thumbnail content (typically an img)
+ * @prop {string} [classes] - Additional CSS classes to apply
+ * @prop {boolean} [isLoading=false] - Is the thumbnail loading?
```

Comment:
Can you expand a little on your arguments for this? The reason I'd like to keep it optional is to allow for the use case of the content inside not being "load-able", but perhaps always present. That is, the loading state doesn't necessarily apply to all uses of this component.

Reply:
If we think generically about _content_ that is not _loadable_ I feel we may dilute the purpose of the `Thumbnail` component, which I would define as a holder of space until an _image_ is loaded.

If that above definition is correct there are only two states: (1) show a spinner (or alternative content) while `isLoading === false` or (2) show the image (no alternative content) when `isLoading === true`.

I think that enforcing the `isLoading` prop could lead developers to decide the correct component. Has it two states? If not, then probably I should use another component to reserve space.

Reply:
> If that above definition is correct there are only two states: (1) show a spinner (or alternative content) while isLoading === false or (2) show the image (no alternative content) when isLoading === true.

I still don't quite agree. Let me illustrate using the immediate use case of the VitalSource book picker.

1. The user enters the book picker, which asks them to paste a link to a VitalSource book. The `Thumbnail` is there to show the cover image of a book. A loading state is not appropriate because the user hasn't taken action to enter a URL yet, so we have "nothing to load" yet. This is when I'd like to render the placeholder.
2. The user enters a URL. Then we go into a loading state.
3. The result of that loading might be that we find a book successfully, in which case we render a cover image as content. There is also the sad path to accommodate, which is likely another form of placeholder. Again, we're not actively trying to load content in this case, so a placeholder is more appropriate than an eternal loading spinner.